### PR TITLE
sword and shield

### PR DESCRIPTION
**Pull request by s-schmitt**
The sword is complete, but the shield still needs a lot of work. 

build: https://mocsarcade.github.io/starjunk/94

### CODE REVIEW

File: source/GameContainer.js [hunk 1]
```diff
@@ -86,6 +86,7 @@ export default class GameContainer extends Pixi.Container {
         }
         this.spawnWaveInterval = 0
         this.difficulty = Reference.DIFFICULTY[0]
+        game.removeChildAt(25)
```
Comment (Phyoog) [inline source/GameContainer.js:89]: What's this for? Removing a child at a specific index isn't necessarily going to remove the same thing every time.

**Comment by s-schmitt** [comment]
I made it where the sword is always placed at that index so I can easily remove it when a player dies or a game over happens

**Comment by Phyoog** [comment]
You could just have the sword itself call `game.removeChild(this)`. Would be a bit less sketchy.

The sword is surprisingly fun to use. Since it stays on once "fired," could we just fire it as soon as it's equipped?

Aesthetically, I like it. What do you think about adding a crossguard?

**Comment by s-schmitt** [comment]
I can change that up real quick and I was actually thinking the sword needed something like that to make it a little more interesting to use.

**Comment by Phyoog** [comment]
Fairly significant bug... you're making the sword invisible, but not actually destroying it. It's still there and anything that hits it is destroyed. It also persists from one game to the next.

**Comment by s-schmitt** [comment]
Yeah that may be a problem

**Comment by s-schmitt** [comment]
Where exactly would I be able to call game.removeChild(this)?

**Comment by Phyoog** [comment]
A quick and simple way would be to do it at the beginning of `changePowerUp()`. You could call `game.removeChild(this.powerUp)` and then `this.powerUp.destroy()` and assign the new powerUp and add it to the game container. I imagine you could just set that in there regardless of what kind of powerUp it is, so no need to even check the weapon type. I don't think so, at least.

**Comment by s-schmitt** [comment]
I will give it a go

**Comment by s-schmitt** [comment]
Yet another road block... that way does not work because the sword itself isn't tied to the powerUp but is just a projectile

**Comment by Phyoog** [comment]
Ehh.... again, not the smoothest, but in the the projectile's update() function, you could add a check like 

```
if (this.projectileType == "sword" && this.shotBy.powerUp.name != "SWORD) {
    game.removeChild(this)
    this.destroy()
}
```

**Comment by codaventure** [comment]
My suggestion is that if you want to really use the projectile class as the sword, that you have sword still be a powerup, and hold a reference to a specific projectile, and if the projectile is ever destroyed, recreate a new projectile (i.e. check for undefined). This would allow you to remove the sword's projectile as soon as the powerup itself is destroyed. If you need more explanation, just let me know.

File: source/GameContainer.js (hunk 1 above)
Comment (codaventure) [inline source/GameContainer.js:89]: Yeah, definitely never do this. Not only is it sketchy, it actually wouldn't work if you believed more than one player would have the sword at once.

File: source/Junkership.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import Sound from "./Sound.js"
 import {PowerUp, PeaShoota, TriShoota, FiveShoota, RapidFire, RapidSprayShot,
     SprayShot, SuperSprayShot, CrazySprayShot, VertSprayShot,
     VertShoota, BFG, Laser, PiercingLaser, SuperLaser, Mine,
-    SuperMine, PaintShot} from "./PowerUp.js"
+    SuperMine, PaintShot, Sword, Shield} from "./PowerUp.js"
```
Comment (codaventure) [inline source/Junkership.js:13]: Looks like when you did the merge conflicts you took the wrong line. Only the weapon that the Junkership starts with is needed. This was to remove dependencies. Thus, with you starting with only Sword right now, this line should read `import { Sword } from "./PowerUp.js"`, and just change it back to PeaShoota in the final commit.

File: source/Junkership.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ export default class Junkership extends Pixi.Sprite {
             RapidSprayShot, SprayShot, SuperSprayShot,
             CrazySprayShot, VertSprayShot, VertShoota,
             BFG, Laser, PiercingLaser, SuperLaser, Mine,
-            SuperMine, PaintShot, PeaShoota]
+            SuperMine, PaintShot, Sword, PeaShoota]
```
Comment (codaventure) [inline source/Junkership.js:39]: As part of removing dependencies, `this.WeaponList` isn't actually used at all, and was removed earlier. You can remove it  again with no issues.

**Comment by s-schmitt** [comment]
Going to redo this pull request. Closing this one.

**Pull request closed without merging**